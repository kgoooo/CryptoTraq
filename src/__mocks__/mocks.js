export const mockStyles = {
	mockAnim: {
		boxShadow: '0 0 3px 6px blue'
	},
	mockBtn:{
		"background": "transparent",
		"border": "2px solid #00FFF5",
		"borderRadius": "8px",
		"color": "#00FFF5",
		"fontSize": "inherit",
		"fontWeight": "300",
		"padding": ".5rem 2.5rem",
		"transition": "all ease-in-out .3s"
	}
};

export const mockMethod = jest.fn();

export const mockCryptoList = [
	{name: "Bitcoin", code: "BTC"},
	{name: "Ethereum", code: "ETH"},
	{name: "Doge", code: "DOGE"}
];

export const mockIntlList = [
	{code: "USD"},
	{code: "CAD"},
	{code: "EUR"},
	{code: "GBP"}
];

export const mockCryptoData = { data: [{
		is_crypto: true,
		name: "crypto1",
		code: "c1"
	}, {
		is_crypto: true,
		name: "crypto2",
		code: "c2"
	}, {
		is_crypto: false,
		name: "fakemoney1",
		code: "fk1"
	}, {
		is_crypto: false,
		name: "fakemoney2",
		code: "fk2"
	}]}

export const mockGraphData = [
	{
		zuluTime: "2018-08-16T19:43:20.221129Z",
		unixTime: 1534439534,
		rate: 623,
	},{
		zuluTime: "2018-08-16T16:42:19.697216Z",
		unixTime: 1534441338,
		rate: 627,
	},{
		zuluTime: "2018-08-16T17:12:14.580348Z",
		unixTime: 1534443072,
		rate: 629,
	},{
		zuluTime: "2018-08-16T17:42:18.458736Z",
		unixTime: 1534444934,
		rate: 632,
	},
];

export const mockGraphDataDesktop = {data: [
		{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		},{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		},{
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}
	]};

export const mockGraphDataMobile = {
	data: [
		{
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}, {
			zuluTime: "20 March 11:45",
			unixTime: 1521546300,
			rate: 601
		}, {
			zuluTime: "20 March 10:35",
			unixTime: 1521542100,
			rate: 623
		}
	]
};
