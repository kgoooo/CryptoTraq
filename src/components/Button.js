import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';

const Button = (props) => {
	return (
		<button
			onClick={props.onClick}
			style={[styles.btn__main, props.buttonAnimation]}
		>{props.buttonContent}</button>
	)
};

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	buttonContent: PropTypes.string.isRequired,
	buttonAnimation: PropTypes.object
};

/*  Button styles for Radium defined here and added below to the component on the export statement. */
const styles = {
	btn__main: {
		background: 'transparent',
		borderRadius: '8px',
		border: `2px solid ${RadiumVars.color.colorPrimaryLightBlue}`,
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontSize: 'inherit',
		fontWeight: '300',
		padding: '.5rem 2.5rem',
		transition: 'all ease-in-out .3s',

		':hover': {
			boxShadow: `0 0 3rem ${RadiumVars.color.colorPrimaryBlue}`
		}
	}
};

export default Radium(Button);
