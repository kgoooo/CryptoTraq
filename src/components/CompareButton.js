import React from 'react';

const CompareButton = (props) => {
	return (
		<div  className="btn--compare">
			<button className="btn btn--light" onClick={props.exchangeSearch}>Track</button>
		</div>
	)
}

export default CompareButton;