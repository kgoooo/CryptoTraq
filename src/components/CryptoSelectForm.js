/*  I refactored this to be a sub component for the new SelectionForm component.
		Once again I realized that to keep the code DRY I needed to refactor the selection
		components to make them more reusable.  This now gets passed into Selection form
		to render the proper content for the select form elements.
		---The main reason why there is still this component and the IntlSelectForm is
		because I couldn't get the crypto.name to render conditionally while being mapped.
*/

import React from 'react';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';


const CryptoSelectForm = (props) =>{
	return (
		<select style={styles.selectForm__form}>
			{props.list.map(crypto => (
				<option key={crypto.code} value={crypto.code}>{crypto.name} â€“ {crypto.code}</option>
			))}
		</select>
	)
};

// Radium styles

const styles = {
	selectForm__form: {
		backgroundColor: RadiumVars.color.colorPrimaryDarkBlue,
		border: `2px solid ${RadiumVars.color.colorPrimaryBlue}`,
		borderRadius: '8px',
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontSize: 'inherit',
		height: '32px',
		outline: 'none',
	},
};


export default Radium(CryptoSelectForm);
