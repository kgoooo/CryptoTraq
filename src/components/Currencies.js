import React from 'react';

const Currencies = (props) => {
	const cryptoList = props.cryptoList;
	const intlList = props.intlList;
	return (
		<div>
      <div>
				<ul>{cryptoList.map(c => <li key={c.code}>{c.name} - {c.code}</li>)}</ul>
        <ul>{intlList.map(i => <li key={i.code}>{i.name}</li>)}</ul>
      </div>
		</div>
	)
}

export default Currencies;