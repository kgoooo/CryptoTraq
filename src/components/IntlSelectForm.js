/*  I refactored this to be a sub component for the new SelectionForm component.
		Once again I realized that to keep the code DRY I needed to refactor the selection
		components to make them more reusable.  This now gets passed into Selection form
		to render the proper content for the select form elements.
*/
/*
	I have added the delay and animation styles to this component, which are passed in
	via props from the CryptoTrack component.
 */
import React from 'react';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';

const IntlSelectForm = props => {
	return (
				<select style={[styles.selectForm__form, props.delay, props.anim]}>
					{props.list.map(intl => (
						<option key={intl.code} value={intl.code}>{intl.code}</option>
					))}
				</select>
	)
};

//  Radium styles
const styles = {
	selectForm__form: {
		backgroundColor: RadiumVars.color.colorPrimaryDarkBlue,
		border: `2px solid ${RadiumVars.color.colorPrimaryBlue}`,
		borderRadius: '8px',
		color: RadiumVars.color.colorPrimaryLightBlue,
		fontSize: 'inherit',
		height: '32px',
		outline: 'none'
	}
};

export default Radium(IntlSelectForm);
