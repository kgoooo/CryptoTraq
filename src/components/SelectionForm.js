/*  I added this component to try and reduce repeated styling and code in general.  This has taken the
		place of the individual 'IntlSelectForm and CryptoSelectForm and made them into little sub components.
	  It makes for more reusable code.
*/

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import RadiumVars from '../RadiumVariables';

const SelectionForm = (props) => {
	return (
		<div style={styles.selectForm}>
			<p style={styles.selectForm__label}>{props.formLabel}</p>
			<form onChange={props.selectionFunc}>
				{props.selectForm}
			</form>
		</div>
	)
};

SelectionForm.propTypes = {
	formLabel: PropTypes.string.isRequired,
	selectionFunc: PropTypes.func.isRequired,
	selectForm: PropTypes.element.isRequired
};

//  Radium styles

const styles = {
	selectForm: {
		display: 'inline-block',
		margin: `${RadiumVars.spacing.largeSize} ${RadiumVars.spacing.largeSize}`,

		'@media (max-width: 500px)': {
			margin: `${RadiumVars.spacing.smallSize} ${RadiumVars.spacing.smallSize}`
		}
	},
	selectForm__label: {
		fontWeight: '300'
	}
};

export default Radium(SelectionForm);
export { SelectionForm };