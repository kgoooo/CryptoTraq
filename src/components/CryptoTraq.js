/*
		Did some cleaning up of imports in this component and added Radium for the arrow
		styling.
		This component had some refactoring when I made the SelectionForm component to
		break up the big form component I had before.  Mainly adding the new components
		and changing the props that are passed in accordingly.
		Button component is also used here now instead of CompareButton.
 */
/*
	I added two loading spinner components to be rendered here.  One will show up when the app
	is initializing, especially if the user is on a slower connection.  The second will show
	when the user has tracked currencies and is waiting for the results.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import Button from './Button';
import SelectionForm from './SelectionForm';
import CryptoSelectForm from './CryptoSelectForm';
import IntlSelectForm from './IntlSelectForm';
import ExchangeResult from './ExchangeResult';
import LoadingSpinner from './LoadingSpinner';
import Graph from './Graph';
import GraphOptionsBar from './GraphOptionsBar';
import RadiumVars from '../RadiumVariables';

const CryptoTraq = (props) => {
  return(
    <div>
        <div>
          <SelectionForm
              formLabel={"Select Cryptocurency:"}
              selectionFunc={props.handleCryptoChange}
              selectForm={<CryptoSelectForm list={props.cryptoList} delay={styles.delay1} anim={styles.select__glow}/>}
          />
          <p style={styles.arrow}>&rarr;</p>
          <SelectionForm
            formLabel={"Select international currency:"}
            selectionFunc={props.handleIntlChange}
						selectForm={<IntlSelectForm list={props.intlList} delay={styles.delay2} anim={styles.select__glow}/>}
					/>
          <Button onClick={props.handleExchangeSearch} buttonContent={"Track currency"} buttonAnimation={styles.trackAnim}/>
        </div>
				<div>
					<GraphOptionsBar handleRangeZeroChange={props.handleRangeZeroChange} rangeContent={props.rangeContent}/>
				</div>
				{props.initializing ?
					<LoadingSpinner
						loading={props.initializing}
						content={"Loading currencies"}
					/> : null}
				{props.loading ?
					<LoadingSpinner
						loading={props.loading}
						content={'Fetching your results, please hold on.'}
					/>: null}
        {props.showingExchangeRate === true ? (
          <ExchangeResult
            crypto={props.selectedCrypto}
            intl={props.selectedIntl}
            rate={props.exchangeRate}
            excError={props.showingExchangeError}
          />
        ) : null}
        <Graph
          cryptoData={props.dataSet}
          crypto={props.selectedCrypto}
					rangeZero={props.rangeZero}
        />
			</div>
  )
};

CryptoTraq.propTypes = {
	cryptoList: PropTypes.array.isRequired,
	dataSet: PropTypes.array.isRequired,
	exchangeRate: PropTypes.number.isRequired,
	initializing: PropTypes.bool.isRequired,
	intlList: PropTypes.array.isRequired,
	loading: PropTypes.bool.isRequired,
	rangeContent: PropTypes.string.isRequired,
	rangeZero: PropTypes.bool.isRequired,
	selectedCrypto: PropTypes.string,
	selectedIntl: PropTypes.string,
	showingExchangeError: PropTypes.bool.isRequired,
	showingExchangeRate: PropTypes.bool.isRequired,
	handleCryptoChange: PropTypes.func.isRequired,
	handleIntlChange: PropTypes.func.isRequired,
	handleExchangeSearch: PropTypes.func.isRequired,
};

//  Radium Styles
/*
		Added selectGlowAnim to add a small animation to the selection element to indicate to the user
		that they should take action there first.  Applied with select__glow
		I added this trackBtnAnimation to add a small scale animation to the track button as the
		3rd step to tracking your conversion rate.  It gets passed in as props and set in the button component
 */

const selectGlowAnim = Radium.keyframes({
	'50%': {boxShadow: `0 0 3px 6px ${RadiumVars.color.colorPrimaryLightBlue}`},
});
const trackBtnAnimation = Radium.keyframes({
	'0': {transform: 'scale(1)'},
	'50%': {
		transform: 'scale(1.1)',
		boxShadow: `0 0 3px 6px ${RadiumVars.color.colorPrimaryLightBlue}`
	},
	'100%': {transform: 'scale(1)'}
}, 'track');

const styles = {
	arrow: {
		display: 'inline-block',
		fontSize: '50px',
		color: RadiumVars.color.colorPrimaryBlue,
		textShadow: `0px 0px 25px ${RadiumVars.color.colorPrimaryBlue}`,

		'@media (max-width: 500px)': {
			margin: '0px'
		}
	},
	select__glow: {
		animationName: selectGlowAnim,
		animationDuration: '1.5s',
	},
	trackAnim: {
		animationName: trackBtnAnimation,
		animationDuration: '1.5s',
		animationDelay: '3.5s'
	},
	delay1: { animationDelay: '0.85s' },
	delay2: { animationDelay: '2.35s'}
};

export default Radium(CryptoTraq);