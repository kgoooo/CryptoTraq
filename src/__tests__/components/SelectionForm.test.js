import React from 'react';
import { shallow } from 'enzyme';
import { SelectionForm } from '../../components/SelectionForm';
import {mockCryptoList, mockIntlList, mockMethod} from '../../__mocks__/mocks';
import { CryptoSelectForm } from '../../components/CryptoSelectForm';
import { IntlSelectForm } from '../../components/IntlSelectForm';

describe('<SelectionForm />', () => {
	
	it('Snapshot: Should render correctly with CryptoSelectForm', () => {
		const wrapper = shallow(<SelectionForm formLabel={'Selection Text'} selectionFunc={mockMethod} selectForm={<CryptoSelectForm list={mockCryptoList}/>}/>)
		expect(wrapper).toMatchSnapshot();
	});

	it('Snapshot: Should render correctly with IntlSelectForm', () => {
		const wrapper = shallow(<SelectionForm formLabel={'Selection Text'} selectionFunc={mockMethod} selectForm={<IntlSelectForm list={mockIntlList}/>}/>)
		expect(wrapper).toMatchSnapshot();
	});

	it('should render label correctly', () => {
		const wrapper = shallow(<SelectionForm formLabel={'Selection Text'} selectionFunc={mockMethod} selectForm={<CryptoSelectForm list={mockCryptoList}/>}/>)
		const paragraphTag = wrapper.find('p');
		expect(paragraphTag.text()).toBe('Selection Text');
	});

	it('should render CryptoSelectForm correctly', () => {
		const wrapper = shallow(<SelectionForm formLabel={'Selection Text'} selectionFunc={mockMethod} selectForm={<CryptoSelectForm list={mockCryptoList}/>}/>)
		expect(wrapper.find('CryptoSelectForm')).toHaveLength(1);
	})
});
