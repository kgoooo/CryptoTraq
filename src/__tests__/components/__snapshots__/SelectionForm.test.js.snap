// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectionForm /> Should render correctly 1`] = `
<div
  style={
    Object {
      "@media (max-width: 500px)": Object {
        "margin": "12px 12px",
      },
      "display": "inline-block",
      "margin": "32px 32px",
    }
  }
>
  <p
    style={
      Object {
        "fontWeight": "300",
      }
    }
  >
    Selection Text
  </p>
  <form
    onChange={[MockFunction]}
  >
    <CryptoSelectForm
      list={
        Array [
          Object {
            "code": "BTC",
            "name": "Bitcoin",
          },
          Object {
            "code": "ETH",
            "name": "Ethereum",
          },
          Object {
            "code": "DOGE",
            "name": "Doge",
          },
        ]
      }
    />
  </form>
</div>
`;

exports[`<SelectionForm /> Snapshot: Should render correctly with CryptoSelectForm 1`] = `
<div
  style={
    Object {
      "@media (max-width: 500px)": Object {
        "margin": "12px 12px",
      },
      "display": "inline-block",
      "margin": "32px 32px",
    }
  }
>
  <p
    style={
      Object {
        "fontWeight": "300",
      }
    }
  >
    Selection Text
  </p>
  <form
    onChange={[MockFunction]}
  >
    <CryptoSelectForm
      list={
        Array [
          Object {
            "code": "BTC",
            "name": "Bitcoin",
          },
          Object {
            "code": "ETH",
            "name": "Ethereum",
          },
          Object {
            "code": "DOGE",
            "name": "Doge",
          },
        ]
      }
    />
  </form>
</div>
`;

exports[`<SelectionForm /> Snapshot: Should render correctly with IntlSelectForm 1`] = `
<div
  style={
    Object {
      "@media (max-width: 500px)": Object {
        "margin": "12px 12px",
      },
      "display": "inline-block",
      "margin": "32px 32px",
    }
  }
>
  <p
    style={
      Object {
        "fontWeight": "300",
      }
    }
  >
    Selection Text
  </p>
  <form
    onChange={[MockFunction]}
  >
    <IntlSelectForm
      list={
        Array [
          Object {
            "code": "USD",
          },
          Object {
            "code": "CAD",
          },
          Object {
            "code": "EUR",
          },
          Object {
            "code": "GBP",
          },
        ]
      }
    />
  </form>
</div>
`;
